libraries/chain/snapshot.cpp

```diff
@@ -386,6 +337,82 @@ void istream_snapshot_reader::return_to_header() {
    clear_section();
 }
 
+istream_json_snapshot_reader::istream_json_snapshot_reader(std::istream& snapshot)
+   :snapshot(snapshot)
+   ,header_pos(snapshot.tellg())
+   ,num_rows(0)
+   ,cur_row(0)
+{
+   std::stringstream my_buffer;
+   my_buffer << snapshot.rdbuf();
```

Comment:
This looks like it reads the entire file in to memory? How much memory does this take for something like an EOS or WAX snapshot? It almost seems like that's going to be impractical.